packages/web-components/src/text-input/text-input.spec.ts

```diff
@@ -0,0 +1,636 @@
+    await expect(element).toHaveJSProperty('appearance', 'underline');
+
+    await element.evaluate((node: TextInput) => {
+      node.appearance = 'filled-darker';
+    });
+    await expect(element).toHaveAttribute('appearance', 'filled-darker');
+    await expect(element).toHaveJSProperty('appearance', 'filled-darker');
+
+    await element.evaluate((node: TextInput) => {
+      node.appearance = 'filled-lighter';
+    });
+    await expect(element).toHaveAttribute('appearance', 'filled-lighter');
+    await expect(element).toHaveJSProperty('appearance', 'filled-lighter');
+  });
+
+  test('should initialize to the `initialValue` property if no value property is set', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+                <fluent-text-input></fluent-text-input>
+            `;
+    });
+
+    const initialValue = await element.evaluate<string, TextInput>(node => node.initialValue);
+
+    await expect(element).toHaveJSProperty('value', initialValue);
+  });
+
+  test('should initialize to the provided `value` attribute if set pre-connection', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+                <fluent-text-input value="foo"></fluent-text-input>
+            `;
+    });
+
+    const element = page.locator('fluent-text-input');
+
+    await expect(element).toHaveJSProperty('value', 'foo');
+  });
+
+  test('should initialize to the provided `value` property if set pre-connection', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ ``;
+
+      const textField = document.createElement('fluent-text-input') as TextInput;
+
+      textField.value = 'foo';
+
+      node.appendChild(textField);
+    });
+
+    await expect(element).toHaveJSProperty('value', 'foo');
+  });
+
+  test('should initialize to the provided `value` attribute if set post-connection', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+                <fluent-text-input></fluent-text-input>
+            `;
+    });
+
+    await element.evaluate(node => {
+      node.setAttribute('value', 'foo');
+    });
+
+    await expect(element).toHaveJSProperty('value', 'foo');
+  });
+
+  test('should hide the label when no default slotted content is provided', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = '<fluent-text-input></fluent-text-input>';
+    });
+
+    await expect(label).toHaveClass(/label__hidden/);
+  });
+
+  test('should hide the label when start content is provided', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+                <fluent-text-input>
+                    <div slot="start"></div>
+                </fluent-text-input>
+            `;
+    });
+
+    await expect(label).toHaveClass(/label__hidden/);
+  });
+
+  test('should hide the label when end content is provided', async () => {
```

Comment:
Not sure if these failed, but our story shows them rendering start and end and label.